util/module.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021 Praetorian Security, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"github.com/go-git/go-git/v5"
+	"fmt"
+	"strings"
+	"os"
+	"errors"
+)
+
+// CloneModule clones a remote git repository over HTTP.
+func CloneModule(dir string, url string) error {
+	// fmt.Printf("git clone %s\n", url)
+	fmt.Printf("Loading new racetrack: %s\n",url)
+
+	_, err := git.PlainClone(dir, false, &git.CloneOptions{

```

Comment:
This will fail with private repository which may need a token, key or credientials in order to clone it.

Reply:
I don't think a GIT client has any way to tell whenever a repository is private instead of just nonexistent. In either case trying to access a private/nonexistent repository with the git cli tool returns a message that says 

`Please make sure you have the correct access rights and the repository exists.`

I agree with you that the errors here are a little generic, but the errors are being passed up directly from the go-git library. 

I did turn off the output from the PlainClone so it might be worth re-enabling that or printing it in case of an error to add some more context to the error.